### PR TITLE
Revamp text tutorial

### PR DESCRIPTION
**Pull request by huiwengoh**
- Updated text tutorial to remove scikeras dependency and use cleanlab's `KerasWrapperSequential` instead
- Text tutorial now demonstrates finding label issues using `CleanLearning` + how to train a model given the issues found
- Add numpy support to cleanlab's Keras wrappers

### CODE REVIEW

File: cleanlab/experimental/keras.py [hunk 1]
```diff
@@ -37,7 +37,9 @@
 
 import tensorflow as tf
 import numpy as np
+import pandas as pd
 from typing import Callable, Optional
+from cleanlab.internal.validation import assert_valid_inputs
 
 
 class KerasWrapperModel:
```
Comment (jwmueller) [inline cleanlab/experimental/keras.py:45]: does the `fit()` method of the classes in here need an updated docstring?

File: docs/source/tutorials/text.ipynb [hunk 2]
```diff
@@ -656,42 +616,22 @@
...
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "When we train the cleanlab-wrapped model, the following operations take place: The original model is trained in a cross-validated fashion to produce out-of-sample predicted probabilities. Then, these predicted probabilities are used to identify label issues, which are then removed from the dataset. Finally, the original model is trained once more on the remaining clean subset of the data.\n"
+    "`CleanLearning` provides a wrapper that can be applied to any scikit-learn compatible model. The resulting model object can be used in the same manner, but it will now train more robustly if the data has noisy labels.\n",
+    "\n",
+    "We can use the same `CleanLearning` object defined above, and since we have already identified the label issues in Section 3 above, we can pass the label issues we found into the `.fit()` call using the `label_issues` argument, which will speed up the model training process as it no longer has to perform cross-validation to first identify the label issues."
```
Comment (jwmueller) [inline docs/source/tutorials/text.ipynb:623]:
```suggestion
    "We can use the same `CleanLearning` object defined above, and  pass the label issues we already computed into `.fit()` via the `label_issues` argument. This accelerates things; if we did not provide the label issues, then they would be recomputed via cross-validation. After that `CleanLearning` simply deletes the examples with label issues and retrains your model on the remaining data."
```

File: docs/source/tutorials/text.ipynb [hunk 3]
```diff
@@ -440,10 +441,9 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "from cleanlab.filter import find_label_issues\n",
-    "\n",
-    "ranked_label_issues = find_label_issues(\n",
-    "    labels=full_labels, pred_probs=pred_probs, return_indices_ranked_by=\"self_confidence\"\n",
+    "print(\n",
+    "    f\"cleanlab found {len(identified_issues)} potential label errors.\\n\"\n",
```
Comment (jwmueller) [inline docs/source/tutorials/text.ipynb:445]:
```suggestion
    "    f\"cleanlab found {len(identified_issues)} potential label errors in the dataset.\\n\"\n",
```

File: docs/source/tutorials/text.ipynb [hunk 4]
```diff
@@ -375,39 +384,29 @@
...
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "An additional benefit of cross-validation is that it facilitates more reliable evaluation of our model than a single training/validation split."
+    "The `find_label_issues` method above will perform cross validation to compute out-of-sample predicted probabilites for each example, which is used to identify label issues.\n",
+    "\n",
+    "This method will return a dataframe that contains the label quality score for each example. The label quality scores are a float between 0 and 1, where a lower score indicate that an example is more likely to be mislabeled. The dataframe also contains a boolean column which specifies if that example is identified as a label error. "
```
Comment (jwmueller) [inline docs/source/tutorials/text.ipynb:409]:
```suggestion
    "This method returns a dataframe containing a label quality score for each example. These numeric scores lie between 0 and 1, where  lower scores indicate examples more likely to be mislabeled. The dataframe also contains a boolean column specifying whether or not each example is identified to have a label issue (indicating it is likely mislabeled)."
```

File: docs/source/tutorials/text.ipynb [hunk 5]
```diff
@@ -106,9 +107,9 @@
    "source": [
     "# Package installation (hidden on docs.cleanlab.ai).\n",
     "# If running on Colab, may want to use GPU (select: Runtime > Change runtime type > Hardware accelerator > GPU)\n",
-    "# Package versions we used: tensorflow==2.9.1 scikeras==0.9.0 scikit-learn==1.1.3 tensorflow_datasets==4.5.2\n",
+    "# Package versions we used: tensorflow==2.9.1 scikit-learn==1.2.0 tensorflow_datasets==4.5.2\n",
```
Comment (jwmueller) [inline docs/source/tutorials/text.ipynb:110]: I don't think you removed scikeras from the !pip install command from the above cell

File: docs/source/tutorials/text.ipynb [hunk 6]
```diff
@@ -42,10 +42,11 @@
     "from cleanlab.classification import CleanLearning\n",
     "\n",
     "cl = CleanLearning(model)\n",
-    "_ = cl.fit(train_data, labels)\n",
-    "label_issues = cl.get_label_issues()\n",
-    "preds = cl.predict(test_data) # predictions from a version of your model \n",
-    "                              # trained on auto-cleaned data\n",
+    "label_issues = cl.find_label_issues()  # label issues identified\n",
```
Comment (jwmueller) [inline docs/source/tutorials/text.ipynb:45]:
```suggestion
    "label_issues = cl.find_label_issues()  # identify potentially mislabeled data\n",
```

File: docs/source/tutorials/text.ipynb [hunk 7]
```diff
@@ -416,22 +415,24 @@
...
-    "print(f\"Cross-validated estimate of log loss: {loss:.3f}\")"
+    "label_issues.head()"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## 4. Use cleanlab to find potential label errors\n"
+    "We can then obtain the subset of identified label errors, and find the index of the top 10 label errors by sorting the label quality scores."
```
Comment (jwmueller) [inline docs/source/tutorials/text.ipynb:425]:
```suggestion
    "We can get the subset of examples flagged with label issues, and also sort by label quality score to find the indices of the 10 most likely mislabeled examples in our dataset."
```

File: cleanlab/experimental/keras.py (hunk 1 above)
Comment (huiwengoh) [reply cleanlab/experimental/keras.py:45]: Yup missed it early, updated!

File: docs/source/tutorials/text.ipynb [hunk 8]
```diff
@@ -314,25 +319,27 @@
...
+    "\n",
+    "<a id=\"section3\"></a>"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Here, we build a simple neural network for classification with TensorFlow and Keras.\n"
+    "Here, we build a simple neural network for classification with TensorFlow and Keras. We will also wrap it with cleanlab's `KerasWrapperSequential` to make it compatible with `CleanLearning`.\n"
```
Comment (jwmueller) [inline docs/source/tutorials/text.ipynb:342]:
```suggestion
    "Here, we build a simple neural network for classification with TensorFlow and Keras. We will also wrap it with cleanlab's `KerasWrapperSequential` to make it compatible with sklearn (and hence`CleanLearning`). Note: you can wrap *any* existing Keras model this way, by just replacing `keras.Sequential` with `KerasWrapperSequential` in your code. \n"
```

File: docs/source/tutorials/text.ipynb [hunk 9]
```diff
@@ -341,8 +348,8 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "def get_net():\n",
-    "    net = tf.keras.Sequential(\n",
+    "def get_nn_model():\n",
+    "    model = KerasWrapperSequential(\n",
```
Comment (jwmueller) [inline docs/source/tutorials/text.ipynb:352]:
```suggestion
    "    model = KerasWrapperSequential(,  # simply replace `keras.Sequential(` with cleanlab's class in this line to make any keras model sklearn-compatible \n",
```

File: docs/source/tutorials/text.ipynb [hunk 10]
```diff
@@ -341,8 +348,8 @@
...
     "        [\n",
```
Comment (jwmueller) [inline docs/source/tutorials/text.ipynb:353]:
```suggestion
    "        [  # the rest of your existing keras code does not need to change at all \n",
```

File: docs/source/tutorials/text.ipynb [hunk 11]
```diff
@@ -351,22 +358,24 @@
...
     "            layers.Dense(num_classes),\n",
     "            layers.Softmax()\n",
-    "        ]\n",
-    "    )  # outputs probability that text belongs to class 1\n",
-    "\n",
-    "    net.compile(\n",
-    "        optimizer=\"adam\",\n",
-    "        loss=tf.keras.losses.SparseCategoricalCrossentropy(),\n",
-    "        metrics=tf.keras.metrics.CategoricalAccuracy(),\n",
+    "        ], # outputs probability that text belongs to class 1\n",
```
Comment (jwmueller) [inline docs/source/tutorials/text.ipynb:361]:
```suggestion
    "        ],  # outputs probability that text belongs to class 1\n",
```

**Review by jwmueller — APPROVED** [review]
Great stuff! Just had minor remaining suggestions to make it more clear to keras users that minimal changes are required from them to use our wrapper

**Review by jwmueller — APPROVED** [review]
